### PR TITLE
PAYOSWXP-120: Adds new log feature: Notification Forward logs are now saved in payone_transaction_forward

### PR DESCRIPTION
**Pull request by amirinterlutions**
PAYOSWXP-120: Adds new log feature: Notification Forward logs are now saved in `payone_transaction_forward`
~~PAYOSWXP-121: Fixes Notification Forward: Now capable of sending notifications to a URL.~~

### CODE REVIEW

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php [hunk 1]
```diff
@@ -4,21 +4,23 @@
...
 use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Messenger\Handler\MessageSubscriberInterface;
 
 class NotificationForwardHandler implements MessageSubscriberInterface
 {
     public function __construct(
         private readonly EntityRepository $notificationForwardRepository,
-        private readonly LoggerInterface $logger
+        private readonly Logger  $logger
```
Comment (rommelfreddy) [inline src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:23]: why did you remove the LoggerInterface?

i think the loggerinterface is better, because some other devs could replace the logger with another logger, if they want. In your case, the constructor will fail, if another logger will be passed.

Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:23]: I need to understand the log status with the `Monolog\Level` in response status.

The Monolog\Logger, with its `addRecord()` method, allows us to dynamically utilize status levels such as `info, error, critical, and warning`.

Using `Psr\Log\LoggerInterface` doesn't seem to achieve the same. In fact, I observed that the Shopware Core Framework and many Shopware plugins use `Monolog`. I assume that this should not pose any issues. I just wanted to ensure that I don't have to write additional code if there is a Shopware package available.

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php [hunk 2]
```diff
@@ -149,4 +161,23 @@ private function buildHeaders(
 
         return $headers;
     }
+
+    private function statusLogger(array $responseInfo, ?string $responseContent, string $id): void
+    {
+        $statusCode = $responseInfo['http_code'];
+
+        $response = new Response($responseContent, $statusCode, $responseInfo);
```
[the remark this answers is not in the document]
Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:169]: Thanks! Yes, I tried to structure it a bit. There's a method `isSuccessful()` that specifically checks if the HTTP code is between 200 and 300 (as you prefer). I used it to log the correct status ($loglevel) for all statuses.

All curl-info looks like this:
```json
{
  "url": "https://payoswxp.ngrok.io/notif.php",
  "content_type": "text/html; charset=UTF-8",
  "http_code": 200,
  "header_size": 197,
  "request_size": 163,
  "filetime": -1,
  "ssl_verify_result": 0,
  "redirect_count": 0,
  "total_time": 0.107368,
  "namelookup_time": 0.00117,
  "connect_time": 0.019795,
  "pretransfer_time": 0.046854,
  "size_upload": 469.0,
  "size_download": 28.0,
  "speed_download": 261.0,
  "speed_upload": 4383.0,
  "download_content_length": -1.0,
  "upload_content_length": 469.0,
  "starttransfer_time": 0.0469,
  "redirect_time": 0.0,
  "redirect_url": "",
  "primary_ip": "3.125.102.39",
  "certinfo": [],
  "primary_port": 443,
  "local_ip": "172.21.0.2",
  "local_port": 54576,
  "http_version": 3,
  "protocol": 2,
  "ssl_verifyresult": 0,
  "scheme": "HTTPS",
  "appconnect_time_us": 46726,
  "connect_time_us": 19795,
  "namelookup_time_us": 1170,
  "pretransfer_time_us": 46854,
  "redirect_time_us": 0,
  "starttransfer_time_us": 46900,
  "total_time_us": 107368
}
```
There are actually no status texts, so I used text from `Response::$statusTexts`.

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php [hunk 3]
```diff
@@ -71,10 +81,10 @@ private function getNotificationForwards(array $ids, Context $context): EntitySe
     }
 
     private function updateResponses(
-        \CurlMultiHandle $multiHandle,
+        \CurlMultiHandle   $multiHandle,
```
[the remark this answers is not in the document]
Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:84]: I think I used the Easy Code Standard from Payone. I should take another look to understand why it wasn't corrected.

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php [hunk 4]
```diff
@@ -110,14 +120,16 @@ private function getForwardRequests(\CurlMultiHandle $multiHandle, EntitySearchR
...
-            $serialize = unserialize($forward->getContent(), []);
-            /** @var array<int, string>|string|false $content */
-            $content = mb_convert_encoding((string) $serialize, 'ISO-8859-1', 'UTF-8');
+            $content = unserialize($forward->getContent(), []);
 
             if (!\is_array($content)) {
                 continue;
             }
 
+            foreach ($content as $key => $value) {
```
[the remark this answers is not in the document]
Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:129]: Ah, okay. So, you actually solved it more easily than I did. I noticed that your commit hasn't been merged into the Master yet, so I'm creating a new branch for the Task `PAYOSWXP-121`. (I'm made the same changes as you.)

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:169]: I corrected the branch name, and further commits are here.
`PAYOSWXP-120` → https://github.com/PAYONE-GmbH/shopware-6/pull/290

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (amirinterlutions) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:129]: I made a pull request: https://github.com/PAYONE-GmbH/shopware-6/pull/289

File: src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php (hunk 1 above)
Comment (rommelfreddy) [reply src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php:23]: ok make sense for me. Thank for clarification

**Comment by rommelfreddy** [comment]
continue in https://github.com/PAYONE-GmbH/shopware-6/pull/290
